api-reference/_hidden/GridBaseColumn/headerFilter/groupInterval.md

```diff
@@ -6,12 +6,165 @@ default: undefined
 ---
 ---
 ##### shortDescription
-Specifies how the header filter combines values into groups.
+Specifies how the header filter combines values into groups. Does not apply if you [specify a custom header filter data source](/Documentation/Guide/UI_Components/DataGrid/How_To/Customize_Header_Filter_Data_Source/#Specify_a_Custom_Data_Source).
 
 ---
-For date columns, set this property to one of the string values. Groups in date columns are hierarchical, and the string value indicates up to which level the hierarchy is formed. The default level is *"day"*, which means that each group has the following structure: *"year" &rarr; "months" &rarr; "days"*.
+For numeric columns, assign a number to this property. This number designates a step with which to generate groups. Column values are classified into these groups.
 
-For numeric columns, assign a number to this property. This number designates a step with which groups should be generated. Column values are classified into these groups.
+#include common-demobutton with {
+    url: "https://js.devexpress.com/Demos/WidgetsGallery/Demo/DataGrid/Filtering/"
+}
+
+For date columns, set this property to one of the accepted string values above. Dates are grouped into a hierarchy, and the string value indicates its lowest level. The default level is *"day"*, which means that the header filter forms the following hierarchy: Year &rarr; Month &rarr; Day. You can disable the hierarchical display if you set the **groupInterval** to **null**. In this case, you also need to implement the column's [calculateFilterExpression](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#calculateFilterExpression) function as follows:
+
+---
+##### jQuery
+
+    <!-- tab: index.js -->
+    $(function() {
+        $("#{widgetName}Container").dx{WidgetName}({
+            // ...
+            columns: [{
+                // ...
+                headerFilter: {
+                    groupInterval: null
+                },
+                calculateFilterExpression(value, operation, target) {
+                    if(value && target === "headerFilter") {
+                        return [this.dataField, operation, new Date(value)];
```

Comment:
I tried to copy this example to demo: https://codepen.io/vconst/pen/vYxpOyo. And, It is not necessary to trap value with new Date(). This line can be changed to:
`return [this.dataField, operation, value];`